File: Diseño de Interfaces/KeepNotes/MainActivity.java
package com.example.keepnotes;

import android.graphics.drawable.AnimatedVectorDrawable;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import java.util.ArrayList;

import static com.example.keepnotes.R.id.fab;

public class MainActivity extends AppCompatActivity
{

    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        final FloatingActionButton fab = findViewById(R.id.fab);
        fab.setImageResource(R.drawable.heart);
        fab.setOnClickListener(new View.OnClickListener()
        {
            public void onClick(View view)
            {
                Snackbar.make(view, "Probando...", Snackbar.LENGTH_LONG).show();
                AnimatedVectorDrawable rotar = (AnimatedVectorDrawable) getDrawable(R.drawable.ad_rotation);
                fab.setImageDrawable(rotar);
                rotar.start();
            }
        });

        final ArrayList<Tarjeta> items = new ArrayList<Tarjeta>();
        items.add(new Tarjeta(R.string.note1));
        items.add(new Tarjeta(R.string.note2));
        items.add(new Tarjeta(R.string.note3));
        items.add(new Tarjeta(R.string.note4));
        items.add(new Tarjeta(R.string.note5));
        items.add(new Tarjeta(R.string.note6));

        final RecyclerView recView = (RecyclerView) findViewById(R.id.recyclerView);

        recView.setHasFixedSize(true);


        CardsAdapter adaptador = new CardsAdapter(items);
        recView.setAdapter(adaptador);
        recView.setItemAnimator(new DefaultItemAnimator());
        recView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));

        adaptador.setOnClickListener(new View.OnClickListener()
        {
            public void onClick(View v)
            {
                Toast.makeText(MainActivity.this, "" + recView.getChildAdapterPosition(v), Toast.LENGTH_SHORT).show();
            }
        });
    }

    public boolean onCreateOptionsMenu(Menu menu)
    {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    public boolean onOptionsItemSelected(MenuItem item)
    {
        int id = item.getItemId();

        if ( id == R.id.action_settings )
        {
            return true;
        }
        if ( id == R.id.action_options )
        {
            return true;
        }
        if ( id == R.id.action_search )
        {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: Diseño de Interfaces/RecyclerView GridLayout y CardsView (Kotlin)/MainActivity.kt
package com.example.travel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.DefaultItemAnimator



class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_cards)

        var items: ArrayList<Card>? = null

            items = ArrayList<Card>()
            items!!.add(Card(R.drawable.whale, "Card 1"))
            items!!.add(Card(R.drawable.whale2,"Card 2"))
            items!!.add(Card(R.drawable.whale3,"Card 3"))
            items!!.add(Card(R.drawable.whale4,"Card 4"))
            items!!.add(Card(R.drawable.whale5,"Card 5"))
            items!!.add(Card(R.drawable.whale6,"Card 6"))
            items!!.add(Card(R.drawable.whale7,"Card 7"))
            items!!.add(Card(R.drawable.whale8,"Card 8"))

            val toolbar = findViewById(R.id.toolbar) as Toolbar
            setSupportActionBar(toolbar)

            val recView = findViewById(R.id.recView) as RecyclerView
            val adaptador = AdaptadorCards(items!!)

            recView.adapter = adaptador
            recView.layoutManager = GridLayoutManager(this, 2)
            recView.itemAnimator = DefaultItemAnimator()
    }
}

